pkg/console/controllers/route/controller.go

```diff
@@ -0,0 +1,211 @@
+)
+
+type RouteSyncController struct {
+	// clients
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	routeClient          routeclientv1.RoutesGetter
+	// names
+	targetNamespace string
+	routeName       string
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+func NewRouteSyncController(
+	// clients
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	routev1Client routeclientv1.RoutesGetter,
+	// informers
+	operatorConfigInformer v1.ConsoleInformer,
+	routeInformer routesinformersv1.RouteInformer,
+	// names
+	targetNamespace string,
+	routeName string,
+	// events
+	recorder events.Recorder,
+) *RouteSyncController {
+	ctrl := &RouteSyncController{
+		operatorConfigClient: operatorConfigClient,
+		routeClient:          routev1Client,
+		targetNamespace:      targetNamespace,
+		routeName:            routeName,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	operatorConfigInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	routeInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		operatorConfigInformer.Informer().HasSynced,
+		routeInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+
+	// configNameFilter := operator.FilterByNames(api.ConfigResourceName)
+	// return operator.New(controllerName, c,
+	// 	operator.WithInformer(operatorConfigInformer, configNameFilter),
+	// )
+}
+
+func (c *RouteSyncController) sync() error {
+	// startTime := time.Now()
+	// klog.V(4).Infof("started syncing route %q (%v)", c.routeName, startTime)
+	// defer klog.V(4).Infof("finished syncing route %q (%v)", c.routeName, time.Since(startTime))
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorsv1.Managed:
+		klog.V(4).Infoln("console is in a managed state: syncing route")
+	case operatorsv1.Unmanaged:
+		klog.V(4).Infoln("console is in an unmanaged state: skipping route sync")
+		return nil
+	case operatorsv1.Removed:
+		klog.V(4).Infoln("console is in a removed state: deleting route")
+		return c.removeRoute()
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+
+	// TODO: is this all we need when syncing the route?
+	//  - anything out of place now?
+	//  - test and see if all things still work?
+	//    - yay e2e!
+	_, _, errReason, err := c.SyncRoute(updatedOperatorConfig)
+
+	// TODO: do we need the "toUpdate" bool?
+	status.HandleProgressingOrDegraded(updatedOperatorConfig, "RouteSync", errReason, err)
+	status.SyncStatus(c.operatorConfigClient, updatedOperatorConfig)
+
```

Comment:
Just comparing to refresh myself, I think we forgot `SyncStatus()` on the `CLIDownloadsController`.